src/otx/algo/utils/xai_utils.py

```diff
@@ -57,28 +44,32 @@ def _process(
         predict_result_per_batch: OTXBatchPredEntitiesSupportXAI,
         label_info: LabelInfoTypes,
     ) -> OTXBatchPredEntitiesSupportXAI:
+        # Extract batch data with proper type handling
+        labels = predict_result_per_batch.labels if predict_result_per_batch.labels is not None else []
+        scores = predict_result_per_batch.scores if predict_result_per_batch.scores is not None else []
+
         saliency_map: list[np.ndarray] = [
             saliency_map.cpu().numpy() if isinstance(saliency_map, torch.Tensor) else saliency_map
-            for saliency_map in predict_result_per_batch.saliency_map
+            for saliency_map in predict_result_per_batch.saliency_map  # type: ignore[union-attr]
```

Comment:
is it necessary to omit all these mypy errors? Will it be possible to remove them after all data entities are updated?